### ISSUE TITLE
Clippy raising unknown attribute errors

### ISSUE THREAD

**Issue by maxburke** [MindFlavor/AzureSDKForRust#135]
When building with clippy 0.0.212 the following errors are raised:

```
error: Usage of unknown attribute
  --> /home/max/.cargo/registry/src/github.com-1ecc6299db9ec823/azure_sdk_storage_core-0.20.2/src/rest_client.rs:57:11
   |
57 | #[clippy::needless_pass_by_value]
   |           ^^^^^^^^^^^^^^^^^^^^^^

error: Usage of unknown attribute
   --> /home/max/.cargo/registry/src/github.com-1ecc6299db9ec823/azure_sdk_storage_core-0.20.2/src/rest_client.rs:236:11
    |
236 | #[clippy::too_many_arguments]
    |           ^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

error: Could not compile `azure_sdk_storage_core`.
```

**Comment by MindFlavor** [comment]
Ah yes, they have been deprecated. I will remove them. I'm taking the occasion to clean up other clippy warnings. This is the dev branch: [https://github.com/MindFlavor/AzureSDKForRust/tree/issue/135/dev](https://github.com/MindFlavor/AzureSDKForRust/tree/issue/135/dev), I'll make a PR once done!

**Comment by MindFlavor** [comment]
Should work now! Thanks for finding it!
